app/routes/cdo/search/basic.js

```diff
@@ -1,19 +1,27 @@
 const { routes, views } = require('../../../constants/search')
+const { routes: formRoutes } = require('../../../constants/forms')
 const { validateUser } = require('../../../api/ddi-index-api/user')
 const { getUser } = require('../../../auth')
 const ViewModel = require('../../../models/cdo/search/basic')
 const searchSchema = require('../../../schema/portal/search/basic')
 const { doSearch } = require('../../../api/ddi-index-api/search')
 const { enforcement } = require('../../../auth/permissions')
 const { addBackNavigation } = require('../../../lib/back-helpers')
+const { licenceNotYetAccepted } = require('../../../lib/route-helpers')
 
 module.exports = [{
   method: 'GET',
   path: routes.searchBasic.get,
   options: {
     auth: { scope: enforcement },
     handler: async (request, h) => {
-      await validateUser(getUser(request))
+      const user = getUser(request)
+
+      await validateUser(user)
+
+      if (await licenceNotYetAccepted(request, user)) {
```

Comment:
would probably put this in separate line, but not going to let it stop a merge.  I guess it works, just have never seen before.